File: trunk/test/compress-spectra.pl
#!/usr/bin/perl -w
use strict;
use warnings;
use MIME::Base64;

my @mz_list;
my @intensity_list;

my $file_base = 'spectra_ORBI';
open(LIST,"$file_base.txt");
while(<LIST>) {
  chomp;
  my ($mz, $intensity) = split(/\s+/);
  push(@mz_list,$mz);
  push(@intensity_list,$intensity);
}
close(LIST);

my $count = $#mz_list + 1;
open(ASCII,">$file_base"."_ascii.encode");
print ASCII $count,"\n";
print ASCII join(" ",@mz_list),"\n";
print ASCII join(" ",@intensity_list),"\n";
close(ASCII);

my $packed_mz = pack("d".$count,@mz_list);
my $packed_intensity = pack("d".$count,@intensity_list);
open(PACK,">$file_base"."_packed.encode");
print PACK $packed_mz;
print PACK $packed_intensity;
close(PACK);

my @unpack_mz = unpack("d".$count, $packed_mz);
my @unpack_intensity = unpack("d".$count, $packed_intensity);
open(UNPACK,">$file_base"."_unpacked.encode");
print UNPACK $count,"\n";
print UNPACK join(" ",@unpack_mz),"\n";
print UNPACK join(" ",@unpack_intensity),"\n";
close(UNPACK);

my $encoded_mz = encode_base64($packed_mz);
my $encoded_intensity = encode_base64($packed_intensity);

open(BASE,">$file_base"."_base64.encode");
print BASE $count,"\n";
print BASE $encoded_mz;
print BASE "__XXXXXX__\n";
print BASE $encoded_intensity;
close(BASE);

File: test/compress-spectra.py
#!/usr/bin/python2.5
import struct
import base64
import array

file_base = 'spectra_ORBI'
f_spectra = open(file_base+'.txt','r')

mz_list = array.array('d')
intensity_list = array.array('d')
for line in f_spectra:
  line = line.strip()
  tokens = line.split(" ")
  mz_list.append(float(tokens[0]))
  intensity_list.append(float(tokens[1]))
f_spectra.close()

count = len(mz_list)
f_ascii = open(file_base+'_ascii.encode','w')
f_ascii.write("%d\n" % count)
f_ascii.write( " ".join([ str(mz) for mz in  mz_list])+"\n" )
f_ascii.write( " ".join([ str(intensity) for intensity in intensity_list])+"\n" )
f_ascii.close()

packed_mz = ''
packed_intensity = ''
for mz in mz_list:
  packed_mz += struct.pack('d',mz)
for intensity in intensity_list:
  packed_intensity += struct.pack('d',intensity)

f_packed = open(file_base+'_packed.encode','w')
f_packed.write( packed_mz )
f_packed.write( packed_intensity )
f_packed.close()

unpacked_mz = struct.unpack('%dd'%count, packed_mz)
unpacked_intensity = struct.unpack('%dd'%count, packed_intensity)
f_unpacked = open(file_base+'_unpacked.encode','w')
f_unpacked.write("%d\n" % count)
f_unpacked.write( " ".join([ str(mz) for mz in  unpacked_mz ])+"\n" )
f_unpacked.write( " ".join([ str(intensity) for intensity in unpacked_intensity ])+"\n" )
f_unpacked.close()

f_base = open(file_base+'_base64.encode','w')
f_base.write("%d\n"%count)
f_base.write(base64.b64encode(packed_mz)+"\n")
f_base.write("__XXXXXX__\n")
f_base.write(base64.b64encode(packed_intensity)+"\n")
f_base.close()
